crates/partition-store/src/partition_store.rs

```diff
@@ -606,6 +609,33 @@ impl PartitionStore {
     pub fn partition(&self) -> &Arc<Partition> {
         self.db.partition()
     }
+
+    pub fn verify_deprecations(&self, key_range: RangeInclusive<PartitionKey>) {
+        // Verify invocation status v1 is unused
+        let Ok(iter) = self.iterator_from::<InvocationStatusKeyV1>(
+            TableScan::FullScanPartitionKeyRange(key_range),
+        ) else {
+            return;
+        };
+
+        if iter.valid() {
+            // Something is here!
+            if let Some(invocation_id) = iter
+                .key()
+                .and_then(|mut k| InvocationStatusKeyV1::deserialize_from(&mut k).ok())
+                .and_then(|k| k.into_inner_ok_or().ok())
+                .map(|(pk, uuid)| InvocationId::from_parts(pk, uuid))
+            {
+                error!(
+                    "Detected an invocation with id {invocation_id} stored using the deprecated format. Cancel or kill the invocation.\nTHIS WILL BECOME AN HARD ERROR IN 1.6"
+                )
```

Comment:
1- Isn't this a big ask for our users? is there a way to auto migrate those invocations instead?
2- What if the server operator is not he owner of the invocation, e.g. restate cloud or a platform team?
3- Is the intention here to warn about a single invocation_id even if the table has many entries?

Side nit: no need for shouting in log messages.